x/ibc/04-channel/keeper/handshake.go

```diff
@@ -169,11 +180,13 @@ func (k Keeper) ChanOpenAck(
 		)
 	}
 
-	// TODO: blocked by #5542
-	// key := sdk.NewKVStoreKey(portID)
-	// if !k.portKeeper.Authenticate(key, portID) {
-	// 	return sdkerrors.Wrap(port.ErrInvalidPort, portID)
-	// }
+	if _, ok := k.GetChannelCapability(ctx, portID, channelID); !ok {
```

Comment:
This should be turned into an `authenticateCapability` call, see [the spec](https://github.com/cosmos/ics/tree/cwgoes/capability-rework/spec/ics-004-channel-and-packet-semantics#opening-handshake):

```typescript
abortTransactionUnless(authenticateCapability(channelCapabilityPath(portIdentifier, channelIdentifier), capability))
```